shared/src/commonMain/kotlin/io.newm.shared/internal/implementations/utilities/MapErrors.kt

```diff
@@ -0,0 +1,28 @@
+package io.newm.shared.internal.implementations.utilities
+
+import io.newm.shared.public.models.error.KMMException
+import kotlin.coroutines.cancellation.CancellationException
+
+suspend fun <T> mapErrorsSuspend(block: suspend () -> T): T {
+    try {
+        return block()
+    } catch (e: KMMException) {
+        throw e
+    } catch (e: CancellationException) {
+        throw e
+    } catch (e: Throwable) {
+        throw KMMException("We encountered a problem.", e)
+    }
+}
+
+fun <T> mapErrors(block: () -> T): T {
+    try {
+        return block()
+    } catch (e: KMMException) {
+        throw e
+    } catch (e: CancellationException) {
+        throw e
+    } catch (e: Throwable) {
+        throw KMMException("We encountered a problem", e)
+    }
+}
```

Comment:
What do you think about this @newmskywalker, can you think of a cleaner alternative?

```
suspend fun <T> runCatchingAsyc(block: suspend () -> T): T {
    return runCatching { block() }
        .getOrElse { e ->
            when (e) {
                is KMMException, is CancellationException -> throw e
                else -> throw KMMException("We encountered a problem: ${e.message}", e)
            }
        }
}
```
```
fun <T> runCatchingSync(block: () -> T): T {
    return runCatching { block() }
        .getOrElse { e ->
            when (e) {
                is KMMException, is CancellationException -> throw e
                else -> throw KMMException("We encountered a problem: ${e.message}", e)
            }
        }
}
```

Reply:
Since the function itself can be called from a coroutine asynchronously, we don't need the runCatchingAsync version of this.